src/cfnlint/rules/resources/elb/Elb.py

```diff
@@ -44,24 +44,48 @@ def check_protocol_value(self, value, path, **kwargs):
 
         return matches
 
-    def check_alb_subnets(self, props, path):
+    def is_application_loadbalancer(self, properties):
+        """ Check if type is application """
+        elb_type = properties.get('Type')
+        if not elb_type or elb_type == 'application':
+            return True
+        return False
+
+    def check_alb_subnets(self, properties, path):
         """ Validate at least two subnets with ALBs"""
         matches = []
-        elb_type = props.get('Type')
-        if elb_type == 'application':
-            subnets = props.get('Subnets')
+        if self.is_application_loadbalancer(properties):
+            subnets = properties.get('Subnets')
             if isinstance(subnets, list):
                 if len(subnets) < 2:
                     path = path + ['Subnets']
                     matches.append(RuleMatch(path, 'You must specify at least two Subnets for load balancers with type "application"'))
-            subnet_mappings = props.get('SubnetMappings')
+            subnet_mappings = properties.get('SubnetMappings')
             if isinstance(subnet_mappings, list):
                 if len(subnet_mappings) < 2:
                     path = path + ['SubnetMappings']
                     matches.append(RuleMatch(path, 'You must specify at least two SubnetMappings for load balancers with type "application"'))
 
         return matches
 
+    def check_loadbalancer_allowed_attributes(self, properties, path):
+        """ Validate loadbalancer attributes per loadbalancer type"""
+        matches = []
+        only_allowed_alb_values = ['idle_timeout.timeout_seconds', 'routing.http2.enabled']
+        only_allowed_nlb_values = ['load_balancing.cross_zone.enabled']
+        loadbalancer_attributes = properties.get('LoadBalancerAttributes')
```

Comment:
Few things I would change.  This just helps just in case someone really poorly formats their template or they use conditions.  What we usually trip up on is someone using a FindInMap or a Condition in the key or in the list.  These safe guards (I think) will help prevent the rule failing when the unusual but necessary may be done.
```python
loadbalancer_attributes = properties.get('LoadBalancerAttributes')
if isinstance(loadbalancer_attributes, list):   # validate the item exists and is a list (just in case)
  key = item.get('Key')   
  value = item.get('Value')
  if isinstance(key, six.string_types) and isinstance(value, (six.string_types, bool)):   # validate key, value are string values
    if self.is_application_loadbalancer(properties):
      if key in only_allowed_nlb_values:
```